### PR TITLE
Add outputFormat as a none-custom param of MapfishPrintProvider

### PR DESCRIPTION
**Pull request by bentrm**
I have the requirement to "print" other formats than pdf via Geoserver printing and thought this may be interesting for upstream as well.
Right now I have problems to configure Geoserver printing to work with other formats than PDF so this has not been tested "live" but only via programmatic tests. I also changed the print-form example to include the outputFormat option but the hosted printing service also fails on delivering the supported formats (as of info.json it should deal with PNG for example).

````
Error while generating PDF:
java.lang.RuntimeException: java.io.IOException: Cannot run program "/usr/bin/convert": error=2, Datei oder Verzeichnis nicht gefunden
	at org.mapfish.print.output.NativeProcessOutputFactory$ImageOutput.print(NativeProcessOutputFactory.java:202)
	at org.mapfish.print.MapPrinter.print(MapPrinter.java:184)
	at org.mapfish.print.servlet.MapPrinterServlet.doCreatePDFFile(MapPrinterServlet.java:336)
	at org.mapfish.print.servlet.MapPrinterServlet.createAndGetPDF(MapPrinterServlet.java:155)
	at org.mapfish.print.servlet.MapPrinterServlet.doGet(MapPrinterServlet.java:86)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:621)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:305)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:224)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:169)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:168)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:98)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:927)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:407)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:987)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:579)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:307)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Cannot run program "/usr/bin/convert": error=2, Datei oder Verzeichnis nicht gefunden
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1047)
	at org.mapfish.print.output.NativeProcessOutputFactory$ImageOutput.createImage(NativeProcessOutputFactory.java:277)
	at org.mapfish.print.output.NativeProcessOutputFactory$ImageOutput.print(NativeProcessOutputFactory.java:193)
	... 22 more
Caused by: java.io.IOException: error=2, Datei oder Verzeichnis nicht gefunden
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:186)
	at java.lang.ProcessImpl.start(ProcessImpl.java:130)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1028)
	... 24 more
`````

If I understand this correctly the printing plugin is trying to convert the generated PDF via ImageMagick command into the desired outputFormat as requested via POST to create.json (described [here](http://www.mapfish.org/doc/print/installation.html#with-image-output-line)).

### CODE REVIEW

**Comment by weskamm** [comment]
> but the hosted printing service also fails on delivering the supported formats

Has been fixed on the server side, the servlet used in the examples now supports image output, tested for jpg and png

**Comment by bentrm** [comment]
The changes seem to be fine then. Thank you. Off to configuring my service correctly next.

**Comment by marcjansen** [comment]
Is this waiting for review? @weskamm, any chance you may want to do it?

**Comment by weskamm** [comment]
I'm having a look at this just right now..

**Comment by bentrm** [comment]
Thanks @weskamm. I think it's ready for review.

File: src/GeoExt/data/MapfishPrintProvider.js [hunk 1]
```diff
@@ -458,6 +494,16 @@ Ext.define('GeoExt.data.MapfishPrintProvider', {
      *
      * @param {Ext.data.Record} layout The record of the layout.
```
Comment (weskamm) [inline src/GeoExt/data/MapfishPrintProvider.js]: needs adjustments from c&p

**Comment by weskamm** [comment]
I just added a minor comment. Otherwise, this looks good to me, thanks for the contribution!

**Comment by bentrm** [comment]
Is this interesting for upstream? Otherwise I will close the PR.

**Comment by marcjansen** [comment]
@weskamm?

**Comment by weskamm** [comment]
Yes i still think it is a nice addition. And as it keeps the PDF format as the default, i dont't see any problems here merging this into upstream and will do so now!

**Pull request merged**